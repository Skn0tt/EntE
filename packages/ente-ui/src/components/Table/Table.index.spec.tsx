/**
 * EntE
 * (c) 2017-present, Simon Knott <info@simonknott.de>
 *
 * This source code is licensed under the GNU Affero General Public License
 * found in the LICENSE file in the root directory of this source tree.
 */

import * as React from "react";
import createTable, { changeSearch, changeSortField } from "./Table";
import { shallow } from "enzyme";

describe("Table", () => {
  const onClick = jest.fn();

  type Person = {
    name: string;
    alter: number;
  };

  const items: Person[] = [
    {
      name: "Bernd",
      alter: 16
    },
    {
      name: "Volker",
      alter: 17
    }
  ];

  const PersonTable = createTable<Person>();

  const comp = shallow(
    <PersonTable
      cellExtractor={i => [i.name, i.alter]}
      headers={["Name", "Alter"]}
      items={items}
      classes={{}}
      keyExtractor={i => i.name}
    />
  );

  it("renders correctly", () => {
    expect(comp).toMatchSnapshot();
  });

  describe("changeSearch", () => {
    it("changes the search", () => {
      const search = "Hallo";
      expect(
        changeSearch(search)({ searchTerm: "jkl", sortField: 1, sortUp: false })
          .searchTerm
      ).toEqual(search);
    });
  });

  describe("changeSortField", () => {
    describe("when same field", () => {
      it("doesnt change the field", () => {
        const field = 5;
        expect(
          changeSortField(field)({
            searchTerm: "jkl",
            sortField: 5,
            sortUp: false
          }).sortField
        ).toEqual(field);
      });
      it("changes the direction", () => {
        const field = 5;
        expect(
          changeSortField(field)({
            searchTerm: "jkl",
            sortField: 5,
            sortUp: false
          }).sortUp
        ).toEqual(true);
      });
    });

    describe("different Field", () => {
      it("does change the field", () => {
        const field = 7;
        expect(
          changeSortField(field)({
            searchTerm: "jkl",
            sortField: 5,
            sortUp: false
          }).sortField
        ).toEqual(field);
      });
      it("keeps direction on true", () => {
        const field = 7;
        expect(
          changeSortField(field)({
            searchTerm: "jkl",
            sortField: 5,
            sortUp: false
          }).sortUp
        ).toEqual(true);
      });
    });
  });
});
