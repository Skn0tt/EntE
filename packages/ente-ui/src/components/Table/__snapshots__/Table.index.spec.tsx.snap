// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Table renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WithStyles(Table)
    cellExtractor={[Function]}
    classes={Object {}}
    headers={
      Array [
        "Name",
        "Alter",
      ]
    }
    items={
      Array [
        Object {
          "alter": 16,
          "name": "Bernd",
        },
        Object {
          "alter": 17,
          "name": "Volker",
        },
      ]
    }
    keyExtractor={[Function]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "cellExtractor": [Function],
      "classes": Object {
        "fab": "Table-fab-2",
        "searchBar": "Table-searchBar-1",
        "table": "Table-table-3",
      },
      "headers": Array [
        "Name",
        "Alter",
      ],
      "items": Array [
        Object {
          "alter": 16,
          "name": "Bernd",
        },
        Object {
          "alter": 17,
          "name": "Volker",
        },
      ],
      "keyExtractor": [Function],
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "cellExtractor": [Function],
        "classes": Object {
          "fab": "Table-fab-2",
          "searchBar": "Table-searchBar-1",
          "table": "Table-table-3",
        },
        "headers": Array [
          "Name",
          "Alter",
        ],
        "items": Array [
          Object {
            "alter": 16,
            "name": "Bernd",
          },
          Object {
            "alter": 17,
            "name": "Volker",
          },
        ],
        "keyExtractor": [Function],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
